scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
```

Comment:
I admit I don't quite understand why Unicode URL is converted to bytes within `Link`.
Was the intent to make it safe? (in that case, we could store the safe_url version of it, using encoding).
Reading `scrapy/tests/test_link.py`, I find [the following](https://github.com/scrapy/scrapy/blob/master/tests/test_link.py#L53) odd:

```
            link = Link(u"http://www.example.com/\xa3")
            self.assertIsInstance(link.url, str)
            self.assertEqual(link.url, b'http://www.example.com/\xc2\xa3')  <-----
```